### PR TITLE
Maistra 2.4 to 1.24.7

### PR DESCRIPTION
**Pull request by twghu**
Update Maistra-2.4 to upstream 1.24.7

### CODE REVIEW

File: source/common/http/conn_manager_utility.cc [hunk 1]
```diff
@@ -305,9 +305,6 @@ void ConnectionManagerUtility::cleanInternalHeaders(
   request_headers.removeEnvoyIpTags();
   request_headers.removeEnvoyOriginalUrl();
   request_headers.removeEnvoyHedgeOnPerTryTimeout();
-  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.sanitize_original_path")) {
```
Comment (jwendell) [inline source/common/http/conn_manager_utility.cc:300]: What's this commit about? Is it cherry-picking an change from another branch?

If it's a cherry-pick, it should contain the original commit metadata (`git cherry-pick`)

Comment (jwendell) [reply source/common/http/conn_manager_utility.cc:300]: JFYI, this is about the last commit "sync conn_manager with upstream"

Comment (dcillera) [reply source/common/http/conn_manager_utility.cc:300]: I can't find this in the upstream. I guess it's a modification Tim has made to fix the error he had  found in ConnectionManagerUtilityTest, but can't say more.

Comment (tedjpoole) [reply source/common/http/conn_manager_utility.cc:300]: This change should be part of [f27b78e1e49851013a6388f2673cc4a7739a0601](https://github.com/maistra/envoy/pull/267/commits/f27b78e1e49851013a6388f2673cc4a7739a0601) which _is_ part of this PR, but which isn't complete.
I'm guessing the lines in this commit got missed out (or more accurately not removed) while merging, due to a bug that Tim was experiencing with CLion?

**Comment by twghu** [comment]
Deprecated for https://github.com/maistra/envoy/pull/268

**Pull request closed without merging**